### ISSUE TITLE
PLEASE RESTART THE SERVER - Always get it

### ISSUE THREAD

**Issue by mykeehu** [Gourieff/sd-webui-reactor#187]
### First, confirm

- [X] I have read the [instruction](https://github.com/Gourieff/sd-webui-reactor/blob/main/README.md) carefully
- [X] I have searched the existing issues
- [X] I have updated the extension to the latest version

### What happened?

After upgrading, the extension keeps asking for a server restart. I've deleted the folder, installed it via UI and even though I restart it all the time, it keeps telling me to restart the server. Reactor loads regardless.

### Steps to reproduce the problem

1. Just upgraded with git pull, nothing more.

### Sysinfo

 {
date:  "Wed Nov 8 23:14:45 2023",
timestamp:  "23:14:47",
uptime:  "Wed Nov 8 23:10:41 2023",
version:  {
app:  "stable-diffusion-webui.git",
updated:  "2023-08-31",
hash:  "5ef669de",
url:  "https://github.com/AUTOMATIC1111/stable-diffusion-webui.git/tree/master"
},
torch:  "2.0.1+cu118 autocast half",
gpu:  {
device:  "NVIDIA GeForce RTX 3090 (1) (compute_37) (8, 6)",
cuda:  "11.8",
cudnn:  8904,
driver:  "546.01"
},
state:  {
started:  "Wed Nov 8 23:14:47 2023",
step:  "0 / 0",
jobs:  "0 / 0",
flags:  "",
job:  "",
text-info:  ""
},
memory:  {
ram:  {
free:  62.33,
used:  1.44,
total:  63.76
},
gpu:  {
free:  20.43,
used:  3.57,
total:  24
},
gpu-active:  {
current:  2.21,
peak:  2.22
},
gpu-allocated:  {
current:  2.21,
peak:  2.22
},
gpu-reserved:  {
current:  2.28,
peak:  2.28
},
gpu-inactive:  {
current:  0.06,
peak:  0.07
},
events:  {
retries:  0,
oom:  0
},
utilization:  0
},
optimizations:  [
0:  "medvram-sdxl"
],
libs:  {
xformers:  "0.0.20",
diffusers:  "0.18.2",
transformers:  "4.30.2"
},
repos:  {
Stable Diffusion:  "[cf1d67a] 2023-03-25",
Stable Diffusion XL:  "[45c443b] 2023-07-26",
CodeFormer:  "[c5b4593] 2022-09-09",
BLIP:  "[48211a1] 2022-06-07",
k_diffusion:  "[ab527a9] 2023-08-12"
},
device:  {
active:  "cuda",
dtype:  "torch.float16",
vae:  "torch.float32",
unet:  "torch.float16"
},
model:  {
configured:  {
base:  "NSFW\realisticVisionV51_v51VAE.safetensors [15012c538f]",
refiner:  "",
vae:  "Automatic"
},
loaded:  {
base:  "M:\!SDModels\NSFW\realisticVisionV51_v51VAE.safetensors",
refiner:  "",
vae:  null
}
},
schedulers:  [
0:  "DDIM" ,
1:  "DPM adaptive" ,
2:  "DPM fast" ,
3:  "DPM++ 2M" ,
4:  "DPM++ 2M Karras" ,
5:  "DPM++ 2M SDE" ,
6:  "DPM++ 2M SDE Exponential" ,
7:  "DPM++ 2M SDE Heun" ,
8:  "DPM++ 2M SDE Heun Exponential" ,
9:  "DPM++ 2M SDE Heun Karras" ,
10:  "DPM++ 2M SDE Karras" ,
11:  "DPM++ 2S a" ,
12:  "DPM++ 2S a Karras" ,
13:  "DPM++ 3M SDE" ,
14:  "DPM++ 3M SDE Exponential" ,
15:  "DPM++ 3M SDE Karras" ,
16:  "DPM++ SDE" ,
17:  "DPM++ SDE Karras" ,
18:  "DPM2" ,
19:  "DPM2 Karras" ,
20:  "DPM2 a" ,
21:  "DPM2 a Karras" ,
22:  "Euler" ,
23:  "Euler a" ,
24:  "Heun" ,
25:  "LMS" ,
26:  "LMS Karras" ,
27:  "PLMS" ,
28:  "Restart" ,
29:  "UniPC"
],
extensions:  [
0:  "ABG_extension (disabled)" ,
1:  "Config-Presets (disabled)" ,
2:  "Hypernetwork-MonkeyPatch-Extension (disabled)" ,
3:  "LDSR (enabled builtin)" ,
4:  "Lora (enabled builtin)" ,
5:  "PBRemTools (disabled)" ,
6:  "SD-latent-mirroring (disabled)" ,
7:  "SadTalker (disabled)" ,
8:  "ScuNET (enabled builtin)" ,
9:  "Stable-Diffusion-WebUI-TensorRT (disabled)" ,
10:  "Stable-Diffusion-Webui-Prompt-Translator (disabled)" ,
11:  "StylePile (disabled)" ,
12:  "SwinIR (enabled builtin)" ,
13:  "a1111-sd-webui-haku-img (disabled)" ,
14:  "a1111-sd-webui-locon (disabled)" ,
15:  "adetailer (enabled)" ,
16:  "batch-face-swap (disabled)" ,
17:  "booru2prompt (enabled)" ,
18:  "canvas-zoom-and-pan (enabled builtin)" ,
19:  "castadi (disabled)" ,
20:  "clip-interrogator-ext (enabled)" ,
21:  "custom-diffusion-webui (disabled)" ,
22:  "danbooru-prompt (enabled)" ,
23:  "ddetailer (disabled)" ,
24:  "depth-image-io-for-SDWebui (disabled)" ,
25:  "depthmap2mask (enabled)" ,
26:  "embedding-inspector (enabled)" ,
27:  "extra-options-section (enabled builtin)" ,
28:  "infinite-zoom-automatic1111-webui (disabled)" ,
29:  "loopback_scaler (disabled)" ,
30:  "mobile (enabled builtin)" ,
31:  "model-keyword (enabled)" ,
32:  "multi-subject-render (disabled)" ,
33:  "openOutpaint-webUI-extension (enabled)" ,
34:  "openpose-editor (enabled)" ,
35:  "posex (enabled)" ,
36:  "prompt-bracket-checker (enabled builtin)" ,
37:  "prompt_translator (disabled)" ,
38:  "sd-canvas-editor (enabled)" ,
39:  "sd-dynamic-thresholding (disabled)" ,
40:  "sd-extension-aesthetic-scorer (disabled)" ,
41:  "sd-extension-system-info (enabled)" ,
42:  "sd-infinity-grid-generator-script (enabled)" ,
43:  "sd-web-ui-quickcss (disabled)" ,
44:  "sd-webui-3d-open-pose-editor (enabled)" ,
45:  "sd-webui-additional-networks (enabled)" ,
46:  "sd-webui-ar (enabled)" ,
47:  "sd-webui-cleaner (enabled)" ,
48:  "sd-webui-controlnet (enabled)" ,
49:  "sd-webui-cutoff (enabled)" ,
50:  "sd-webui-fabric (enabled)" ,
51:  "sd-webui-faceswaplab (disabled)" ,
52:  "sd-webui-freeu (disabled)" ,
53:  "sd-webui-gelbooru-prompt (enabled)" ,
54:  "sd-webui-inpaint-anything (enabled)" ,
55:  "sd-webui-llul (disabled)" ,
56:  "sd-webui-lora-block-weight (enabled)" ,
57:  "sd-webui-model-converter (disabled)" ,
58:  "sd-webui-model-mixer (disabled)" ,
59:  "sd-webui-modelscope-text2video (disabled)" ,
60:  "sd-webui-multiple-hypernetworks (disabled)" ,
61:  "sd-webui-openpose-editor (enabled)" ,
62:  "sd-webui-photopea-embed (enabled)" ,
63:  "sd-webui-reactor (enabled)" ,
64:  "sd-webui-regional-prompter (enabled)" ,
65:  "sd-webui-segment-anything (enabled)" ,
66:  "sd-webui-supermerger (disabled)" ,
67:  "sd_auto_fix (disabled)" ,
68:  "sd_lama_cleaner (enabled)" ,
69:  "sd_web_ui_preset_utils (disabled)" ,
70:  "sdweb-easy-prompt-selector (disabled)" ,
71:  "sdweb-merge-block-weighted-gui (disabled)" ,
72:  "seamless-tile-inpainting (disabled)" ,
73:  "stable-diffusion-webui-aesthetic-gradients (disabled)" ,
74:  "stable-diffusion-webui-aesthetic-image-scorer (disabled)" ,
75:  "stable-diffusion-webui-cafe-aesthetic (disabled)" ,
76:  "stable-diffusion-webui-composable-lora (enabled)" ,
77:  "stable-diffusion-webui-conditioning-highres-fix (disabled)" ,
78:  "stable-diffusion-webui-daam (disabled)" ,
79:  "stable-diffusion-webui-dataset-tag-editor (enabled)" ,
80:  "stable-diffusion-webui-depthmap-script (enabled)" ,
81:  "stable-diffusion-webui-embedding-editor (disabled)" ,
82:  "stable-diffusion-webui-embedding-merge (disabled)" ,
83:  "stable-diffusion-webui-instruct-pix2pix (disabled)" ,
84:  "stable-diffusion-webui-model-toolkit (enabled)" ,
85:  "stable-diffusion-webui-promptgen (enabled)" ,
86:  "stable-diffusion-webui-rembg (enabled)" ,
87:  "stable-diffusion-webui-text2prompt (disabled)" ,
88:  "stable-diffusion-webui-tokenizer (enabled)" ,
89:  "stable-diffusion-webui-two-shot (disabled)" ,
90:  "stable-diffusion-webui-visualize-cross-attention-extension (disabled)" ,
91:  "ultimate-upscale-for-automatic1111 (disabled)"
],
platform:  {
arch:  "AMD64",
cpu:  "Intel64 Family 6 Model 183 Stepping 1, GenuineIntel",
system:  "Windows",
release:  "Windows-10-10.0.19045-SP0",
python:  "3.10.11"
},
crossattention:  "xformers",
backend:  "",
pipeline:  ""
}

### Relevant console log

```Shell
venv "I:\Stable-Diffusion-Automatic\stable-diffusion-webui\venv\Scripts\Python.exe"
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Version: v1.6.0
Commit hash: 5ef669de080814067961f28357256e8fe27544f4
Fetching updates for midas...
Checking out commit for midas with hash: 1645b7e...
Looking in indexes: https://pypi.org/simple, https://pypi.ngc.nvidia.com
Requirement already satisfied: onnxruntime-gpu in i:\stable-diffusion-automatic\stable-diffusion-webui\venv\lib\site-packages (1.13.1)
Requirement already satisfied: coloredlogs in i:\stable-diffusion-automatic\stable-diffusion-webui\venv\lib\site-packages (from onnxruntime-gpu) (15.0.1)
Requirement already satisfied: flatbuffers in i:\stable-diffusion-automatic\stable-diffusion-webui\venv\lib\site-packages (from onnxruntime-gpu) (23.5.26)
Requirement already satisfied: numpy>=1.21.6 in i:\stable-diffusion-automatic\stable-diffusion-webui\venv\lib\site-packages (from onnxruntime-gpu) (1.23.5)
Requirement already satisfied: packaging in i:\stable-diffusion-automatic\stable-diffusion-webui\venv\lib\site-packages (from onnxruntime-gpu) (23.0)
Requirement already satisfied: protobuf in i:\stable-diffusion-automatic\stable-diffusion-webui\venv\lib\site-packages (from onnxruntime-gpu) (3.20.2)
Requirement already satisfied: sympy in i:\stable-diffusion-automatic\stable-diffusion-webui\venv\lib\site-packages (from onnxruntime-gpu) (1.11.1)
Requirement already satisfied: humanfriendly>=9.1 in i:\stable-diffusion-automatic\stable-diffusion-webui\venv\lib\site-packages (from coloredlogs->onnxruntime-gpu) (10.0)
Requirement already satisfied: mpmath>=0.19 in i:\stable-diffusion-automatic\stable-diffusion-webui\venv\lib\site-packages (from sympy->onnxruntime-gpu) (1.2.1)
Requirement already satisfied: pyreadline3 in i:\stable-diffusion-automatic\stable-diffusion-webui\venv\lib\site-packages (from humanfriendly>=9.1->coloredlogs->onnxruntime-gpu) (3.4.1)
ReActor preheating... Device: CUDA

        +---------------------------------+
        --- PLEASE, RESTART the Server! ---
        +---------------------------------+
Installing pyqt5 requirement for depthmap script
Launching Web UI with arguments: --xformers --medvram-sdxl --no-half-vae --ckpt-dir M:\!SDModels --lora-dir I:\!Lora --ui-config-file 1my/ui-config-my.json --embeddings-dir I:\!Embeddings --hypernetwork-dir I:\!Hypernetworks --gfpgan-dir I:\GFPGAN --esrgan-models-path I:\ESRGAN --realesrgan-models-path I:\RealESRGAN --vae-dir I:\VAE --ui-settings-file 1my/config-my.json --autolaunch
[-] ADetailer initialized. version: 23.11.0, num models: 23
Loading booru2prompt settings
[AddNet] Updating model hashes...
100%|█████████████████████████████████████████████████████████████████████████████| 679/679 [00:00<00:00, 67904.92it/s]
[AddNet] Updating model hashes...
100%|█████████████████████████████████████████████████████████████████████████████| 679/679 [00:00<00:00, 84878.62it/s]
2023-11-08 23:10:44,446 - ControlNet - INFO - ControlNet v1.1.416
ControlNet preprocessor location: I:\annotator
2023-11-08 23:10:44,521 - ControlNet - INFO - ControlNet v1.1.416
Loading FABRIC v0.6.3
23:10:45 - ReActor - STATUS - Running v0.5.0-b1
Loading weights [15012c538f] from M:\!SDModels\NSFW\realisticVisionV51_v51VAE.safetensors
Creating model from config: I:\Stable-Diffusion-Automatic\stable-diffusion-webui\configs\v1-inference.yaml
[tag-editor] Settings has been read from config.json
Running on local URL:  http://127.0.0.1:7860
```

**Comment by Gourieff** [comment]
Perhaps, another extension reinstalls smth that ReActor requires 
Does it work despite of the warning?

**Comment by mykeehu** [comment]
Good question, because Reaktor has checked the installation every time before and is doing the same now, but as I see it finds all files OK and still asks for a reboot.

**Comment by Gourieff** [comment]
This message appears only if ReActor detects that some of dependencies are not installed and install them

**Comment by nyukers** [comment]
> This message appears only if ReActor detects that some of dependencies are not installed and install them
**No, Reactor doesnt't install necessary lib:**
ValueError: This ORT build has ['TensorrtExecutionProvider', 'CUDAExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['TensorrtExecutionProvider', 'CUDAExecutionProvider', 'CPUExecutionProvider'], ...)

**Comment by Inmanguo** [comment]
same here:
always get when everything seems satisfied:        
 +---------------------------------+
        --- PLEASE, RESTART the Server! ---
        +---------------------------------+

and it's not working:
ValueError: This ORT build has ['TensorrtExecutionProvider', 'CUDAExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['TensorrtExecutionProvider', 'CUDAExecutionProvider', 'CPUExecutionProvider'], ...)

**Comment by Inmanguo** [comment]
> > This message appears only if ReActor detects that some of dependencies are not installed and install them
> > **No, Reactor doesn't install necessary lib:**
> > ValueError: This ORT build has ['TensorrtExecutionProvider', 'CUDAExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['TensorrtExecutionProvider', 'CUDAExecutionProvider', 'CPUExecutionProvider'], ...)

Have you solved this problem?

**Comment by Gourieff** [comment]
I think, you have another extension that installs the old version of onnxruntime library every time you launch A1111
ReActor needs ORT-GPU or ORT 1.16.1 or higher

**Comment by Gourieff** [comment]
**ReActor installs ORT-GPU 1.16.1 or higher (if you have CUDA) or ORT 1.16.1 or higher (if you don't have CUDA)**

> ValueError: This ORT build has ['TensorrtExecutionProvider', 'CUDAExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['TensorrtExecutionProvider', 'CUDAExecutionProvider', 'CPUExecutionProvider'], ...)

This error appears **ONLY** if you have ORT < 1.16.1 - that can be installed by any other extension
Provide `pip list` for your A1111's VENV

**Comment by Inmanguo** [comment]
the extensions list, just in case

a1111-sd-webui-tagcomplete 
multidiffusion-upscaler-for-automatic1111
 sd-webui-controlnet 
sd-webui-infinite-image-browsing 
sd-webui-lora-block-weight 
sd-webui-model-converter 
sd-webui-openpose-editor 
sd-webui-prompt-all-in-one 
sd-webui-reactor 
sd-webui-segment-anything 
sd-webui-supermerger 
stable-diffusion-webui-localization-zh_Hans
stable-diffusion-webui-model-toolkit 
stable-diffusion-webui-wd14-tagger 
ultimate-upscale-for-automatic1111 
LDSR 
Lora 
ScuNET 
SwinIR 
canvas-zoom-and-pan 
extra-options-section mobile 
prompt-bracket-checker

**Comment by Gourieff** [comment]
> onnxruntime 1.16.2

What GPU do you have?
Please provide the full log from the beginning to the moment when the restart message appears

**Comment by Inmanguo** [comment]
> > onnxruntime 1.16.2
> 
> What GPU do you have? Please provide the full log from the beginning to the moment when the restart message appears

GPU: RTX3090

full log: 

Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Version: v1.6.0-2-g4afaaf8a
Commit hash: 4afaaf8a020c1df457bcf7250cb1c7f609699fa7
ReActor preheating... Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl
Requirement already satisfied: onnxruntime-gpu in e:\sd-webui\python\lib\site-packages (1.14.1)
Requirement already satisfied: coloredlogs in e:\sd-webui\python\lib\site-packages (from onnxruntime-gpu) (15.0.1)
Requirement already satisfied: flatbuffers in e:\sd-webui\python\lib\site-packages (from onnxruntime-gpu) (23.3.3)
Requirement already satisfied: numpy>=1.21.6 in e:\sd-webui\python\lib\site-packages (from onnxruntime-gpu) (1.23.5)
Requirement already satisfied: packaging in e:\sd-webui\python\lib\site-packages (from onnxruntime-gpu) (23.1)
Requirement already satisfied: protobuf in e:\sd-webui\python\lib\site-packages (from onnxruntime-gpu) (4.25.0)
Requirement already satisfied: sympy in e:\sd-webui\python\lib\site-packages (from onnxruntime-gpu) (1.11.1)
Requirement already satisfied: humanfriendly>=9.1 in e:\sd-webui\python\lib\site-packages (from coloredlogs->onnxruntime-gpu) (10.0)
Requirement already satisfied: mpmath>=0.19 in e:\sd-webui\python\lib\site-packages (from sympy->onnxruntime-gpu) (1.3.0)
Requirement already satisfied: pyreadline3 in e:\sd-webui\python\lib\site-packages (from humanfriendly>=9.1->coloredlogs->onnxruntime-gpu) (3.4.1)
Device: CUDA

        +---------------------------------+
        --- PLEASE, RESTART the Server! ---
        +---------------------------------+
Launching Web UI with arguments: --theme dark --xformers --api --autolaunch

#let me know if you need anything else,thx

**Comment by Gourieff** [comment]
@Inmanguo thanks for the log
I got the reason of this behavior
ReActor didn't reinstall any old version of ORT or ORT-GPU to the newest one
(It checked that new version is required but didn't install it afterwards)
Fixed https://github.com/Gourieff/sd-webui-reactor/commit/fda2ae3fa1ef41c38b9d8214d833ecac15d964f9

**Comment by Inmanguo** [comment]
> @Inmanguo thanks for the log I got the reason of this behavior ReActor didn't reinstall any old version of ORT or ORT-GPU to the newest one (It checked that new version is required but didn't install it afterwards) Fixed [fda2ae3](https://github.com/Gourieff/sd-webui-reactor/commit/fda2ae3fa1ef41c38b9d8214d833ecac15d964f9)

It works. Amazing job! Thank you!
